pydbus/xml_generator.py

```diff
@@ -0,0 +1,107 @@
+"""Automatic XML documentation generator."""
+import inspect
+import sys
+
+from xml.etree import ElementTree
+
+
+PROPERTY_EMITS_SIGNAL = "org.freedesktop.DBus.Property.EmitsChangedSignal"
+
+
+# Python 2 treats them as methods, Python 3 as functions
+ismethod = inspect.ismethod if sys.version_info[0] == 2 else inspect.isfunction
+
+
+def verify_arguments(function):
+	"""Verify that the function is correctly defined."""
+	args, vargs, kwargs, defaults = inspect.getargspec(function)
+	if defaults is not None:
+		raise ValueError(
+			"Default values are not allowed for method "
+			"'{}'".format(function.__name__))
+	# TODO: Check if vargs or kwargs are actually allowed in dbus.
```

Comment:
Yeah, dbus_context won't get passed if somebody is using **kwargs. But we need to handle the case when it is stated explicitely in the function's signature (like def MyMethod(self, sth1, sth2, dbus_context=None):)